main/src/library/IpAddr.flix

```diff
@@ -38,5 +40,27 @@ mod IpAddr {
         case _ => unreachable!()
     }
 
+    ///
+    /// Converts an `IpAddr` into an `Array[Int8, r]`
+    ///
+    /// an Ipv4 is 4 bytes and Ipv6 is 16 bytes
+    ///
+    pub def toByteArray(rc: Region[r], ip: IpAddr): Array[Int8, r] \ r =
```

Comment:
@CadeMichael fyi since I know you are making changes in this space. We may want to consider `toByteList`/`fromByteList` instead of just `toBytes`/`fromBytes` to help distinguish.